src/lib/Graphs/IO/NodeSpecJSON.ts

```diff
@@ -1,20 +1,19 @@
-export type LinkJSON = { node: number; socket: string };
+import { NodeCategory } from '../../Nodes/NodeCategory';
 
 export type InputSocketSpecJSON = {
-  name: string,
-  valueType: string,
-  defaultValue: string;
+  name: string;
+  valueType: 'flow' |'string'|'number'|'boolean',
+  defaultValue?: string | number | boolean
```

Comment:
This should always be a string I believe.

Reply:
Hmm, it isn't in some of the examples, there are zeros and bools when I looked. It would make sense to match the type of the valueType though